source/SH3/graphics/texture.cpp

```diff
@@ -9,52 +9,340 @@
  *  @author Jesse Buhagiar
  */
 #include <SH3/graphics/texture.hpp>
+#include <SH3/system/assert.hpp>
 #include <SH3/system/log.hpp>
 #include <SH3/arc/vfile.hpp>
 
+#include <algorithm>
 #include <cassert>
-
+#include <fstream>
+#include <functional>
+#include <limits>
 
 using namespace sh3_graphics;
 
+#pragma pack(push, 1)
+struct rgba
+{
+    std::uint8_t r, g, b, a;
+};
+
+struct rgba16
+{
+    uint8_t r : 5;
+    uint8_t g : 5;
+    uint8_t b : 5;
+    uint8_t a : 1;
+};
+
+struct rgb
+{
+    std::uint8_t r, g, b;
+};
+
+struct bgra
+{
+    std::uint8_t b, g, r, a;
+};
+
+struct bgr
+{
+    std::uint8_t b, g, r;
+};
+
+struct tga_header
+{
+    static constexpr std::uint8_t TYPE_RGB24 = 2;
+    static constexpr std::uint8_t FLAGS_FLIP = 0x20;
+
+    std::uint8_t id_size = 0;               /**< Size of the ID field that follows this header (usually 0) */
+    std::uint8_t colormap = 0;              /**< Is this image paletted (THIS MUST ALWAYS BE 0 FOR US!) */
+    std::uint8_t type = TYPE_RGB24;         /**< Type of image (ALWAYS 2 FOR RGB24!) */
+    std::uint8_t unused[5] = {0, 0, 0, 0, 0};
+
+    std::uint16_t x_origin = 0;             /**< Co-ordinate for first X value */
+    std::uint16_t y_origin = 0;             /**< Co-ordinate for first Y value */
+    std::uint16_t width = 0;
+    std::uint16_t height = 0;
+    std::uint8_t bpp = 24;                  /**< Bits per pixel */
+    std::uint8_t flags = FLAGS_FLIP;
+};
+
+#pragma pack(pop)
+
+/**
+ *  Dump a texture to a TARGA/TGA file
+ *
+ *  @param width - The Width of this texture.
+ *  @param height - The Height of this texture.
+ *  @param data - Pixel data
```

Comment:
I actually meant further qualifying "pixel data" in the comment.
Passing the format (RGB8, RGB5A1, ...) would also work very well, though it would require the function to be extended to support other formats - or simply refuse to work with non-RGB formats.